modules/core/arrow-core-extensions/src/main/kotlin/arrow/core/extensions/function0.kt

```diff
@@ -92,6 +94,16 @@ interface Function0Monad : Monad<ForFunction0> {
 
   override fun <A, B> Function0Of<Either<A, B>>.select(f: Kind<ForFunction0, (A) -> B>): Kind<ForFunction0, B> =
     fix().fun0Select(f)
+
+  override suspend fun <A> MonadContinuation<ForFunction0, *>.bindStrategy(fa: Function0Of<A>): BindingStrategy<ForFunction0, A> =
+    BindingStrategy.Strict(fa.fix()())
+
+  override val fx: MonadFx<ForFunction0>
+    get() = object : MonadFx<ForFunction0> {
```

Comment:
This will create a new instance on every call. Is that what we want? I believe we should be catching it instead, use something like by lazy or having the object on the companion.

Reply:
Interfaces `by lazy` won't work here. For `* -> *` we can cache this on the companion, however.